src/qnetvo/network_ansatz.py

```diff
@@ -257,18 +204,57 @@ def collect_wires(network_nodes):
 
         :raises ValueError: If the same wire is used in two different nodes in ``network_nodes``.
         """
-        ansatz_wires = list(map(lambda node: qml.wires.Wires(node.wires), network_nodes))
-        all_wires = qml.wires.Wires.all_wires(ansatz_wires)
-        unique_wires = qml.wires.Wires.unique_wires(ansatz_wires)
-
-        # two nodes cannot prepare a state on the same wire
-        if not all_wires.tolist() == unique_wires.tolist():
-            raise ValueError(
-                "One or more wires are not unique. Each node must contain unique wires."
-            )
+
+        wires_objs = list(map(qml.wires.Wires, wires_lists))
+        all_wires = qml.wires.Wires.all_wires(wires_objs)
+
+        if check_unique:
+            unique_wires = qml.wires.Wires.unique_wires(wires_objs)
+
+            # two nodes cannot prepare a state on the same wire
+            if not all_wires.tolist() == unique_wires.tolist():
+                raise ValueError(
+                    "One or more wires are not unique. Each node must contain unique wires."
+                )
 
         return all_wires
 
+    @staticmethod
+    def check_cc_causal_structure(cc_wires_layers, cc_wires_out_layers):
+        """Verifies that the classical communication is causal.
+
+        Note that ``cc_wires_out`` describes the classical communication senders while ``cc_wires`` describe
+        classical communication receivers.
+        All network ansatzes must have a causal structure where nodes output their classical communication in
+        layers that precede the nodes that use that classical communication.
```

Comment:
Yes, once a node measures a wire to obtain a measurement value, that measurement value is available to all downstream nodes and should not be modified. Any number of nodes can use the measurement result because it is classical and can be copied.